### PR TITLE
change show-all default to true

### PR DESCRIPTION
**Pull request by deads2k**
Kube made their printers hide terminated pods by default.  Since terminated pods are very interesting (did it fail or exit normally, show me those logs, etc), that really hurts usability.  This pull changes the default.

The implementation is a little weird.  We can't change the default inside the godeps code because that would change the behavior of our `kubectl` symlink.  Instead, after the commands are registered, we walk them looking for the bad flag and change its default value, current value, and description.

@kargakis 
@smarterclayton Want to change it for embedded `kubectl` too?

### CODE REVIEW

File: pkg/cmd/cli/cli.go [hunk 1]
```diff
@@ -150,6 +151,25 @@ func NewCommandCLI(name, fullName string, in io.Reader, out, errout io.Writer) *
...
 
+// changeSharedFlagDefaults changes values of shared flags that we disagree with.  This can't be done in godep code because
+// that would change behavior in our `kubectl` symlink. Defend each change.
+// 1. show-all - the most interesting pods are terminated/failed pods.  We don't want to exclude them from printing
+func changeSharedFlagDefaults(rootCmd *cobra.Command) {
+	cmds := []*cobra.Command{rootCmd}
+
+	for i := 0; i < len(cmds); i++ {
+		currCmd := cmds[i]
+		cmds = append(cmds, currCmd.Commands()...)
```
Comment (kargakis) [inline pkg/cmd/cli/cli.go]: Not a big fan of appending to a slice while iterating over it...

**Comment by kargakis** [comment]
+1, LGTM

File: pkg/cmd/cli/cli.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/cli/cli.go]: > Not a big fan of appending to a slice while iterating over it...

I only see this and the recursive solution.  Is there another simple iterative solution I'm just not seeing?

**Comment by smarterclayton** [comment]
Let's leave embedded kubectl as is

On Oct 7, 2015, at 10:42 AM, David Eads <notifications@github.com> wrote:

Kube made their printers hide terminated pods by default. Since terminated
pods are very interesting (did it fail or exit normally, show me those
logs, etc), that really hurts usability. This pull changes the default.

The implementation is a little weird. We can't change the default inside
the godeps code because that would change the behavior of our kubectl
symlink. Instead, after the commands are registered, we walk them looking
for the bad flag and change its default value, current value, and
description.

@kargakis <https://github.com/kargakis>
@smarterclayton <https://github.com/smarterclayton> Want to change it for
embedded kubectl too?
------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/openshift/origin/pull/4993
Commit Summary

   - change show-all default to true

File Changes

   - *M* pkg/cmd/cli/cli.go
   <https://github.com/openshift/origin/pull/4993/files#diff-0> (20)

Patch Links:

   - https://github.com/openshift/origin/pull/4993.patch
   - https://github.com/openshift/origin/pull/4993.diff

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4993>.

**Comment by deads2k** [comment]
> Let's leave embedded kubectl as is

Just for clarity, that means: "I'm ok with this weird looking patch"?

File: pkg/cmd/cli/cli.go (hunk 1 above)
Comment (kargakis) [inline pkg/cmd/cli/cli.go]: > I only see this and the recursive solution. Is there another simple iterative solution I'm just not seeing?

I would prefer recursion but I don't feel strongly though.

**Comment by smarterclayton** [comment]
Yes

File: pkg/cmd/cli/cli.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/cli/cli.go]: > I would prefer recursion but I don't feel strongly though.

Habit.  I trust iterative solutions more than recursive ones.

**Comment by deads2k** [comment]
[merge]

File: pkg/cmd/cli/cli.go (hunk 1 above)
Comment (kargakis) [inline pkg/cmd/cli/cli.go]: > Habit. I trust iterative solutions more than recursive ones.

That's why I don't feel strongly. In the one side you have recursion, in the other side you edit the slice while iterating:)

**Pull request merged**